docs/_plugins/include_example.rb

```diff
@@ -0,0 +1,79 @@
+require 'liquid'
+require 'pygments'
+
+module Jekyll
+  class IncludeExampleTag < Liquid::Tag
+    
+    def initialize(tag_name, markup, tokens)
+      @markup = markup
+      super
+    end
+ 
+    def render(context)
+      site = context.registers[:site]
+      config_dir = (site.config['code_dir'] || '../examples/src/main').sub(/^\//,'')
+      @code_dir = File.join(site.source, config_dir)
+
+      clean_markup = @markup.strip
+      @file = File.join(@code_dir, clean_markup)
+      @lang = clean_markup.split('.').last
+
+      code = File.open(@file).read.encode("UTF-8")
+      code = select_lines(code)
+ 
+      Pygments.highlight(code, :lexer => @lang)
+    end
+ 
+    # Trim the code block so as to have the same indention, regardless of their positions in the
+    # code file.
+    def trim_codeblock(lines)
+      # Select the minimum indention of the current code block.
+      min_start_spaces = lines
+        .select{ |l| l.strip.size !=0 }
```

Comment:
nit: if we still want to follow Spark code style, it should be `select {` instead of `select{`